schema.go

```diff
@@ -121,15 +122,20 @@ func schemaOf(model reflect.Type) *Schema {
 //
 // The function panics if Node contains more leaf columns than supported by the
 // package (see parquet.MaxColumnIndex).
-func NewSchema(name string, root Node) *Schema {
+func NewSchema(name string, root Node, source ...tagSource) *Schema {
+	var ts tagSource = defaultTagSource{}
+	if len(source) > 0 {
+		ts = source[0]
+	}
```

Comment:
I would like to ask for a different approach on this API, the proposed change has two downsides:
- We are introducing a breaking change because the signature of the function changes. It's possible that it won't impact most programs because call sites of `parquet.NewSchema` would still compile, but it could break applications that are referencing the function by value.
- The function takes a variable number of arguments, but it only interprets the first tag source; this is easy to misuse and likely to cause subtle bugs in applications where developers would have intuitively expected that if the function accepts multiple tag source it would use them all.

I think we can accept changing the signature of the function because we are pre-1.0, but we need to come up with a more robust mechanism for configuring the schema. For example, can we have a variable list of schema options so we can potentially expand the configuration set in the future?

Reply:
@achille-roussel 
 Agreed, accepting variadic arguments and only interpreting the first one is ugly and can be confusing sometimes. 

> For example, can we have a variable list of schema options so we can potentially expand the configuration set in the future?

Doesn't this have same shortcomings as current implementation? We will still be looking for a single value of tag source while accepting variable number of inputs.

After reading you comment, since we are fine with breaking the api. We can actually require passing of something like a  `SchemaOptions` struct

```go
type SchemaOptions struct {
	TagSource TagSource
}

func NewSchema(name string, root Node, options SchemaOptions) 
```

To avoid external TagSource implementation we add something like this to `TagSource`

```
type TagSource interface {
	...
	isTagSource()
}
```

We are still future proof but, we will not be designing for the future we don't know yet. 

What do you think ?

Reply:
done 5b30793

I am passing `SchemaOptions` struct. See how it looks and let know, I don't mind exposing `SchemaOption` as a functional variadic api for passing the tag source, the concrete  `SchemaOptions`  struct looks good to me for now.